proof_range.go

```diff
@@ -509,8 +556,9 @@ func (t *ImmutableTree) getRangeProof(keyStart, keyEnd []byte, limit int) (proof
 					Height:  node.subtreeHeight,
 					Size:    node.size,
 					Version: node.version,
-					Left:    nil,
-					Right:   node.rightHash,
+					// Need this information to know where we came from while constructing a DeepSubTree
```

Comment:
I think this should not be necessary to send over the wire. You should be able to compute this locally, no?

Reply:
I tried computing it locally earlier, and reproduced it again here: https://github.com/celestiaorg/iavl/pull/4/files

However, in `computeRootHash` inside `PathToLeaf`, the hash computed through `pin.Hash(hash)` is not always the same as what `pin.Left` contains and that leads to inconsistent hashing. Found removing this optimization to work, so incorporated it for now. 

Please correct me if there's a different way you can think of/something obvious I'm missing here.

Reply:
If `pin.Hash(hash)` returns different things than `pin.Left` then shouldn't the Merkle proof be impossible to verify in the first place? I think something doesn't add up.